mypy/checkexpr.py

```diff
@@ -1664,6 +1664,14 @@ def check_callable_call(
 
         See the docstring of check_call for more information.
         """
+        # Check implicit calls to deprecated class constructors.
+        # Only the non-overload case is handled here. Overloaded constructors are handled
+        # separately during overload resolution. `callable_node` is `None` for an overload
+        # item so deprecation checks are not duplicated.
+        if isinstance(callable_node, RefExpr) and isinstance(callable_node.node, TypeInfo):
+            self.chk.check_deprecated(callable_node.node.get_method("__new__"), context)
+            self.chk.check_deprecated(callable_node.node.get_method("__init__"), context)
```

Comment:
This feels like the wrong place for this. But I don't remember where would be better. Where is overloaded `__init__` checked for deprecation?

Reply:
Thanks for digging into this!